### ISSUE TITLE
Add reduction='none' automatically to loss functions when passed from torch.nn

### ISSUE THREAD

**Issue by maurapintor** [pralab/secml-torch#174]
The current attacks support using loss functions directly passed from pytorch, however they have reduction='mean' by default. This does not allow trackers to work properly. The reduction attribute should be fixed internally (and possibly outputting a warning when this is changed internally).
